### PR TITLE
Felix version conflicts fixed AND option to disable plugins

### LINKED ISSUE #70: Felix Plugins : Option to enable and disable easily
## Currently, we have: 
 - Maven builds any plugin that is configured to build and places the jars to bundles directory
 - Felix registers any plugin that is under the bundles directory

## What we need:
 - Some way of explicitly telling Felix to ignore some plugins in bundles directory

## Future Work
The next version would be to make maven ignore the plugins that are not used. But it is not needed as of now.

### LINKED ISSUE #76: Felix Version issue
Currently, we have multiple versions of felix and osgi in classpath.
[pom.xml](https://github.com/USCDataScience/sparkler/blob/master/pom.xml)
```
        <felix.framework.version>2.0.4</felix.framework.version>
        <felix.core.version>1.4.0</felix.core.version>
        <felix.main.version>5.4.0</felix.main.version>
```

## ToDo:
- Clean this mess - make it easy to upgrade
- set it to the latest stable version

### Notes: 
- `org.osgi` has the specification and `org.apache.felix.*` package has the implementation
- Double check class names for imports, variable declarations and initialization 

### p.s. : 
Caught issues while trying to fix #70

### PR DESCRIPTION
**Pull request by thammegowda**
Changelog
- Felix versions fixed. felix 5.4 and osgi 1.4 were incompatible. Now using osgi 6.0
- No caching of bundles - edit configs to reflect changes for the immediate next run
- option to enable and disable plugins

```yaml
# List of activated plugins
plugins.active:
    - urlfilter.regex
    # - fetcher.jbrowser
```
Closes #70 
Closes #76

### CODE REVIEW

**Comment by thammegowda** [comment]
@karanjeets please  review

File: conf/sparkler-default.yaml [hunk 1]
```diff
@@ -70,11 +73,19 @@ fetcher.server.delay: 1000
 # Discouraged to Modify unless specifically required.
 plugins.bundle.directory: ${project.parent.basedir}${file.separator}${project.bundles.directory}
 
+plugins.cache.directory: felix-cache
```
Comment (karanjeets) [inline conf/sparkler-default.yaml]: As discussed, we don't need this anymore. I will remove this configuration property.

File: pom.xml [hunk 2]
```diff
@@ -38,9 +38,8 @@
         <hbase.version>1.2.1</hbase.version>
         <solr.version>6.4.0</solr.version>
         <slf4j.version>1.7.12</slf4j.version>
-        <felix.framework.version>2.0.4</felix.framework.version>
-        <felix.core.version>1.4.0</felix.core.version>
-        <felix.main.version>5.4.0</felix.main.version>
+        <osgi.version>6.0.0</osgi.version>
+        <felix.version>5.4.0</felix.version>
```
Comment (karanjeets) [inline pom.xml]: Awesome!!

File: sparkler-app/src/main/scala/edu/usc/irds/sparkler/service/PluginService.scala [hunk 3]
```diff
@@ -125,37 +136,68 @@ class PluginService(job:SparklerJob) {
...
 
     // Start the Felix Framework
-    serviceLoader.start()
-    serviceLoader.getBundleContext.getBundles.foreach(b => {
-      println(s"Bundle Found: ${b.getSymbolicName}" )
-    })
+    loader.start()
+  }
+
+  object BundleLoader {
```
Comment (karanjeets) [inline sparkler-app/src/main/scala/edu/usc/irds/sparkler/service/PluginService.scala]: +1

**Comment by karanjeets** [comment]
@thammegowda Great work! I have reviewed it. Will merge it soon.

**Pull request merged**